keras/layers/attention/attention_test.py

```diff
@@ -221,7 +224,132 @@ def test_attention_compute_mask_returns_correct_tensor_with_valid_mask(
         valid_mask = np.array([True, False, True])
         mask = [valid_mask, np.array([False, True, False])]
         computed_mask = layer.compute_mask(inputs=dummy_inputs, mask=mask)
+        if backend.backend() == "torch" and computed_mask.is_cuda:
+            computed_mask = computed_mask.cpu().numpy()
         self.assertTrue(
             np.array_equal(computed_mask, valid_mask),
             "compute_mask did not return the correct mask tensor",
         )
+
+    def test_attention_compute_mask_returns_correct_tensor_with_all_true_mask(
+        self,
+    ):
+        layer = layers.Attention()
+        dummy_inputs = [np.ones((2, 3, 4)), np.ones((2, 4, 4))]
+        valid_mask = np.array([True, True, True])
+        mask = [valid_mask, np.array([True, True, True])]
+        computed_mask = layer.compute_mask(inputs=dummy_inputs, mask=mask)
+        if backend.backend() == "torch" and computed_mask.is_cuda:
+            computed_mask = computed_mask.cpu().numpy()
+        expected_mask = np.array([True, True, True])
+        self.assertTrue(
+            np.array_equal(computed_mask, expected_mask),
+            "compute_mask did not return the correct mask tensor",
+        )
+
+    def test_attention_compute_mask_returns_correct_tensor_with_all_false_mask(
+        self,
+    ):
+        layer = layers.Attention()
+        dummy_inputs = [np.ones((2, 3, 4)), np.ones((2, 4, 4))]
+        valid_mask = np.array([False, False, False])
+        mask = [valid_mask, np.array([False, False, False])]
+        computed_mask = layer.compute_mask(inputs=dummy_inputs, mask=mask)
+        if backend.backend() == "torch" and computed_mask.is_cuda:
+            computed_mask = computed_mask.cpu().numpy()
+        expected_mask = np.array([False, False, False])
+        self.assertTrue(
+            np.array_equal(computed_mask, expected_mask),
+            "compute_mask did not return the correct mask tensor",
+        )
+
+    def test_attention_compute_mask_with_tolerance_1e_3(self):
+        layer = layers.Attention()
+        dummy_inputs = [np.ones((2, 3, 4)), np.ones((2, 4, 4))]
+        valid_mask = np.array([1.0, 0.0, 1.0], dtype=float)
+        mask = [valid_mask, np.array([0.0, 1.0, 0.0], dtype=float)]
+        computed_mask = layer.compute_mask(inputs=dummy_inputs, mask=mask)
+        if backend.backend() == "torch" and computed_mask.is_cuda:
+            computed_mask = computed_mask.cpu().numpy()
```

Comment:
I think you can just use `ops.convert_to_numpy` everywhere. It will work with all backends (no need for `if backend == ...`) and it will take care of moving to CPU.

Reply:
Thank you for your feedback, @fchollet 
Done :)